### ISSUE TITLE
False-positive for `opening_brace ` rule with tuples

### ISSUE THREAD

**Issue by fbp93** [realm/SwiftLint#3574]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The `opening_brace` rule matches a closure parameter in a tuple

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ Pods/SwiftLint/swiftlint lint test.swift
Linting Swift files at paths test.swift
Linting 'test.swift' (1/1)
/Users/bot/dev/swiftlint_test/test.swift:13:24: warning: Opening Brace Spacing Violation: Opening braces should be preceded by a single space and on the same line as the declaration. (opening_brace)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run swiftlint version to be sure)?
0.43.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
CocoaPods
* Paste your configuration file:

```yml
only_rules:
    - opening_brace
```

* Which Xcode version are you using (check `xcodebuild -version`)?
12.4
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
func foo(_ a: (String, () -> Void)) {}

self.foo(
  (
     "String parameter",
     { "Do something here" } // Here the rule triggers the violation
   )
)

```
